src/tls.rs

```diff
@@ -0,0 +1,194 @@
+//! Connection helper.
+use tokio::io::{AsyncRead, AsyncWrite};
+
+use tungstenite::{
+    client::uri_mode, error::Error, handshake::client::Response, protocol::WebSocketConfig,
+};
+
+use crate::{client_async_with_config, domain, IntoClientRequest, WebSocketStream};
+
+pub use crate::stream::MaybeTlsStream;
+
+/// A TLS connector that can be used when establishing TLS connections.
+#[non_exhaustive]
+pub enum TlsConnector {
```

Comment:
brainstorm: maybe `Connector` name would be more appropriate when one variant is `Plain`?

Reply:
That's a very good idea. Not only does it repeat the module name (`tls::TlsConnector`), but it probably would raise questions what a plain TLS connector is ought to be.

I'll rename to simply `Connector`.